src/services/expences.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+class Expences {
+  constructor() {
+    this.expenses = [];
+  }
+
+  reset() {
+    this.expenses = [];
+  }
+
+  getAll(userId, category, from, to) {
+    const filterByUserId = (expense) => {
+      return !userId || expense.userId === Number(userId);
+    };
+
+    const filterByCategory = (expense) => {
+      return !category || category.includes(expense.category);
+    };
+
+    const filterByTime = (expense) => {
+      const expenseDate = new Date(expense.spentAt);
+      const fromDate = from ? new Date(from) : null;
+      const toDate = to ? new Date(to) : null;
+
+      return (!fromDate || expenseDate >= fromDate)
+        && (!toDate || expenseDate <= toDate);
+    };
+
+    this.expenses = this.expenses
```

Comment:
Using multiple filter functions is inefficient. Consider combining the filter functions into a single filter function.